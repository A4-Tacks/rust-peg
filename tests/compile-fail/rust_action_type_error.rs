extern crate peg;

peg::parser!(grammar foo() for str {
    rule foo() -> u32 = "a" { "a" } //~ ERROR
});

fn main() {}
