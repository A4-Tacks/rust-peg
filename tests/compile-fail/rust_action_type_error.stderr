error[E0308]: mismatched types
 --> $DIR/rust_action_type_error.rs:4:31
  |
4 |     rule foo() -> u32 = "a" { "a" } //~ ERROR
  |                               ^^^ expected u32, found reference
  |
  = note: expected type `u32`
             found type `&'static str`

For more information about this error, try `rustc --explain E0308`.
